ovs/extensions/storageserver/storagedriverconfig/base.py

```diff
@@ -0,0 +1,213 @@
+        return f(*args, **kwargs)
+    return wrap
+
+
+class VolumeDriverConfigOption(object):
+    """
+    Config option read out by 'volumedriver_fs', '--config-help-markdown'
+    """
+    _options = []
+
+    def __init__(self, component, key, default_value, dynamically_reconfigurable, remarks):
+        #  type: (str, str, any, bool, str) -> None
+        """
+        Initialize a new option
+        Format of the command is '| component | key | default value | dynamically reconfigurable | remarks |'
+        :param component: Volumedriver component
+        :type component: str
+        :param key: Option key
+        :type key: str
+        :param default_value: Default value
+        :param dynamically_reconfigurable:
+        :param remarks:
+        """
+        self.component = component
+        self.key = key
+        self.default_value = default_value
+        self.dynamically_reconfigurable = dynamically_reconfigurable
+        self.remarks = remarks
+
+    @classmethod
+    @ensure_options
+    def get_all_options(cls):
+        # type: () -> List[VolumeDriverConfigOption]
+        """
+        Get all possible options
+        :return: A list with all options
+        :rtype: List[VolumeDriverConfigOption]
+        """
+        return cls._options
+
+    @classmethod
+    @ensure_options
+    def get_option_by_component_and_key(cls, component, key):
+        # type: (str, str) -> Union[VolumeDriverConfigOption, None]
+        """
+        Retrieve an option by component and key
+        Returns None when no option was found
+        :param component: Component to retrieve
+        :type component: str
+        :param key: Key to retrieve from component
+        :type key: str
+        :return: Option if found else none
+        :rtype: VolumeDriverConfigOption
+        """
+        try:
+            return filter(lambda option: option.key == key and option.component == component, cls._options)[0]
+        except IndexError:
+            return None
+
+    @classmethod
+    @ensure_options
+    def get_options_by_component(cls, component):
+        # type: (str) -> List[VolumeDriverConfigOption]
+        """
+        Retrieve all options for a component
+        :param component: Component to retrieve options for
+        :type component: str
+        :return: All options associated with the component
+        :rtype: List[VolumeDriverConfigOption]
+        """
+        return filter(lambda option: option.component == component, cls._options)
+
+    @classmethod
+    def _cache_options(cls):
+        """
+        Cache all possible options
+        :return: None
+        :rtype: NoneType
+        """
+        options = []
+        markdown = check_output(['volumedriver_fs', '--config-help-markdown'])
+        for line in markdown.split('\n|')[2:]:
+            options.append(cls.from_option_line('|' + line))
+        cls._options = options
+
+    @classmethod
+    def from_option_line(cls, line):
```

Comment:
Not an ideal function name either tbh
`parse_line`, `read_line`, use a verb to show what it does